kratos/geometries/coupling_master_slave.h

```diff
@@ -0,0 +1,452 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Tobias Teschemacher
+//  contributors:    Pooyan Dadvand
+//                   Philipp Bucher
+//
+
+#if !defined(KRATOS_COUPLING_MASTER_SLAVE_H_INCLUDED )
+#define  KRATOS_COUPLING_MASTER_SLAVE_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class CouplingMasterSlave
+ * @ingroup KratosCore
+ * @brief The CouplingMasterSlave acts as topology for faces having a pointer
+ *        to a surface geometry and a list of BrepEdge s. Those edges
+ *        define the trimming of the respective surface.
+ */
+template<class TPointType> class CouplingMasterSlave
```

Comment:
@tteschemacher and @pooyan-dadvand  Just to mention that what I do for the ContactStructuralMechanicsApplication is define a coomon condition to all contact conditions is define a paired condition https://github.com/KratosMultiphysics/Kratos/blob/master/applications/ContactStructuralMechanicsApplication/custom_conditions/paired_condition.h

I am not against this, but I would like to know exactly how this works?, this geometry contains N geometries?, being one of them the "reference condition" (master/slave depending of your definition)?

Reply:
The idea is very similiar to yours. However it does not relate the conditions but directly the geomtries. Thus, we can use the standard constructors for conditions and elements. And don't have to do Set/GetValue operations.

The access GetGeometry() will provide the geometry of the Master. With GetGeometryPart(index) we can then access to any arbitrary part, genereally the slave. (this can handle more than one couplings, or also embedded sliding geometries, ...)

Reply:
For me OK, that that would imply to refactor many things in my app, but I will not block as long as this makes completely sense